hydra_base/lib/template/__init__.py

```diff
@@ -1213,6 +1213,55 @@ def get_templatetype(type_id, include_parent_data=True, **kwargs):
 
     return inherited_templatetype
 
+@required_perms("edit_template")
+def clone_templatetype(type_id, **kwargs):
+    """
+        Clone a specific template type by ID. Creates a copy of the template type
+        with all its typeattrs. The clone will have new IDs and timestamps but
+        identical content otherwise, except the name which will be modified to be unique.
+    """
+
+    #First get the DB entry
+    parent_templatetype = db.DBSession.query(TemplateType).filter(
+        TemplateType.id == type_id).options(noload(TemplateType.typeattrs)).one()
+
+    cloned_templatetype = TemplateType()
+    for col in parent_templatetype.__table__.columns:
+        if col.name in ['id', 'cr_date', 'updated_at']:
+            continue
```

Comment:
The `cr_date` attr is checked for in the `test_templates::test_clone_templatetype` test but is omitted here.